### DISCUSSION TITLE
Development with multiple members on same database

### DISCUSSION THREAD

**Discussion by prateek-ec** [prisma/prisma#13468] [Q&A]
Hello,

First of all, thank you for the library and the support.

I am currently facing a situation where we have 2 developers working on the same PostgreSQL database.
Let's say dev1 and dev2 are working on 2 different branches b1 and b2, respectively. Both are connected to the same database and schema.
Situation 1:
1. If dev1 creates a new table T1 and dev2 creates another table T2. How do they go on doing their development work without impacting one another ? I tried the following case:
Step 1: dev1 pushes table T1 using `db push` first before creating any migration files
Step 2: dev2 starts development and pushes table T2 using `db push`

In this case, table T1 gets deleted from the database since the last schema file synced was from dev2.
Similarly, if dev2 runs the `db pull` command, then it removes changes for model T2 and adds model T1 to the schema file after syncing with the database.

Here are some of the answers I am looking for:
1. Is this a good practice concerning using Prisma?
We can do this activity using different databases and migrating/syncing files after merging the branches and do
`migrate deploy` on our staging and production databases.
I have read the following documents and I am still confused about how to go on ahead with it.
[Syncing between team members](https://www.prisma.io/dataguide/managing-databases/syncing-development-databases-between-team-members)
[Team development](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/team-development)
[Patching production](https://www.prisma.io/docs/guides/database/patching-production)
2. Is there any other way I can do this using a single database and schema for multiple developers?

Any help will be highly appreciated.
Regards

**Comment by pantharshit00** [comment]
Hello @prateek-ec,

Sorry for very late reply here. For the use case you are describing here, I don't think migrate will be a good fit. Migrate can't deal with divergent changes without manually fixing the structure if a drift is detected. I would suggest if you want to use migrate, you can have disposable docker containers for development which your team can recreate and reseed anytime if a divergent change happens.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
